addon/-private/system/model/model.js

```diff
@@ -120,8 +120,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
     @type {Boolean}
     @readOnly
   */
-  hasDirtyAttributes: Ember.computed('currentState.isDirty', function() {
-    return this.get('currentState.isDirty');
+  hasDirtyAttributes: Ember.computed('currentState', function() {
+    return this.get('_internalModel.currentState.isDirty');
```

Comment:
How did you test this? Because we have an `ember-try` scenario for ember.js 1.13 and it looks like the test passes on Travis and AppVeyor.